sierra_adapter/sierra_item_merger/src/test/scala/uk/ac/wellcome/platform/sierra_item_merger/dynamo/MergedSierraRecordDaoTest.scala

```diff
@@ -0,0 +1,191 @@
+      "should return a future of None if the record does not exist in DynamoDB") {
+      val mergedSierraRecord = MergedSierraRecord(id = "b1111",
+                                                  maybeBibData = None,
+                                                  itemData = Map(),
+                                                  version = 1)
+
+      Scanamo.put(dynamoDbClient)(tableName)(mergedSierraRecord)
+
+      whenReady(mergedSierraRecordDao.getRecord("b88888")) { record =>
+        record shouldBe None
+      }
+    }
+
+    it(
+      "returns a failed future with the underlying exception if the DynamoDB read fails") {
+      val dynamoDbClient = mock[AmazonDynamoDB]
+      val expectedException = new RuntimeException("AAAAAARGH!")
+      when(dynamoDbClient.getItem(any[GetItemRequest]))
+        .thenThrow(expectedException)
+
+      val mergedSierraRecordDaoMockedDynamoClient =
+        new MergedSierraRecordDao(dynamoDbClient, DynamoConfig(tableName))
+
+      val future =
+        mergedSierraRecordDaoMockedDynamoClient.getRecord("b88888")
+
+      whenReady(future.failed) { ex =>
+        ex shouldBe expectedException
+      }
+    }
+  }
+
+  describe("update a merged sierra record") {
+    it(
+      "should insert a new merged sierra record if it doesn't already exist in DynamoDB") {
+      val id = "b1111"
+      val mergedSierraRecord = MergedSierraRecord(
+        id = id,
+        maybeBibData = None,
+        itemData = Map(
+          "i111" -> SierraItemRecord(id = "i111",
+                                     data = "something",
+                                     modifiedDate = "2001-01-01T01:01:30Z",
+                                     bibIds = List("b1111"))),
+        version = 0
+      )
+
+      whenReady(mergedSierraRecordDao.updateRecord(mergedSierraRecord)) { _ =>
+        Scanamo
+          .get[MergedSierraRecord](dynamoDbClient)(tableName)('id -> id)
+          .get shouldBe Right(
+          mergedSierraRecord.copy(version = 1)
+        )
+      }
+    }
+
+    it(
+      "should update a merged sierra record if it already exists and has a higher version") {
+      val id = "b1111"
+      val mergedSierraRecord = MergedSierraRecord(id = id,
+                                                  maybeBibData = None,
+                                                  itemData = Map(),
+                                                  version = 1)
+      val newerMergedSierraRecord = mergedSierraRecord.copy(version = 2)
+
+      Scanamo.put(dynamoDbClient)(tableName)(mergedSierraRecord)
+
+      whenReady(mergedSierraRecordDao.updateRecord(newerMergedSierraRecord)) {
+        _ =>
+          Scanamo
+            .get[MergedSierraRecord](dynamoDbClient)(tableName)('id -> id)
+            .get shouldBe Right(
+            newerMergedSierraRecord.copy(version = 3)
+          )
+      }
+    }
+
+    it(
+      "should update a merged sierra record if it already exists and has a the same version") {
```

Comment:
Is this the correct behaviour?

I would think that if you try to put an MSR with the same version as an existing database row, your update gets rejected – to prevent two instances trying to PUT competing updates. Isn’t this why we have the version field?

But I’m 👍 on the test case, checking this boundary is definitely important.

Reply:
I think this is definitely wrong, if the versions are the same it means that another process has updated the record while we were working.

Reply:
I've pushed the logic for incrementing the version in the dao, so that means this case should be correct. The name of the test is probably ambiguous though

Reply:
I've been thoroughly persuaded this is the correct behaviour!

Reply:
Right, so I think I’ve convinced myself the test is correct, it’s the description that’s confusing. But I’m no longer convinced this test does anything interestingly different from the last one.